packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs

```diff
@@ -121,6 +121,22 @@ private IEnumerable<MethodBodyStatement> GetStackVariablesForProtocolParamConver
                         statements.Add(UsingDeclare("content", BinaryContentHelperSnippets.FromEnumerable(parameter), out var content));
                         declarations["content"] = content;
                     }
+                    else if (parameter.Type.IsFrameworkType && !parameter.Type.Equals(typeof(BinaryData)))
+                    {
+                        if (parameter.Type.Equals(typeof(string)))
```

Comment:
Can this case be listed outside the if scope? since it specifies type as `string`, so it won't be `BinaryData` for sure.